kasa/httpclient.py

```diff
@@ -43,27 +46,23 @@ async def post(
     ) -> Tuple[int, Optional[Union[Dict, bytes]]]:
         """Send an http post request to the device."""
         response_data = None
-        cookies = None
-        if cookies_dict:
-            cookies = httpx.Cookies()
-            for name, value in cookies_dict.items():
-                cookies.set(name, value)
-        self.client.cookies.clear()
+        self._last_url = url
+        self.client.cookie_jar.clear()
         try:
             resp = await self.client.post(
                 url,
                 params=params,
                 data=data,
                 json=json,
                 timeout=self._config.timeout,
-                cookies=cookies,
+                cookies=cookies_dict,
                 headers=headers,
             )
-        except httpx.ConnectError as ex:
+        except (aiohttp.ServerDisconnectedError, aiohttp.ClientOSError) as ex:
```

Comment:
We need to retry on `ServerDisconnectedError` since the server is allowed to close the connection at any time
https://datatracker.ietf.org/doc/html/rfc2616#section-8.1.4

This can be done in a followup PR